src/main/scala/com/allantl/jira4s/v2/domain/WorkLogPayLoad.scala

```diff
@@ -0,0 +1,67 @@
+package com.allantl.jira4s.v2.domain
+
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+
+case class WorkLogPayLoad(comment: Option[String], started: String, timeSpentSeconds: Int)
+
+object WorkLogPayLoad {
+  implicit val issueDecoder: Decoder[WorkLogPayLoad] = deriveDecoder
+  implicit val issueEncoder: Encoder[WorkLogPayLoad] = deriveEncoder
+}
+
+case class WorkLogCreateResponse(id: String, self: String)
+
+object WorkLogCreateResponse {
+  implicit val issueDecoder: Decoder[WorkLogCreateResponse] = deriveDecoder
+  implicit val issueEncoder: Encoder[WorkLogCreateResponse] = deriveEncoder
+}
+
+sealed trait AdjustEstimate
+
+case class NewEstimate(newEstimate: String) extends AdjustEstimate
+
+case object LeaveEstimate extends AdjustEstimate
+
+case class ManualEstimate(newEstimate: String) extends AdjustEstimate
+
+case object AutoEstimate extends AdjustEstimate
+
+object Converter {
```

Comment:
Type class for this feels overkill IMO. Just put the url conversion inside each method in`WorklogClient`. Or as a private def should be fine too.